src/sqlfluff/testing/rules.py

```diff
@@ -141,4 +141,12 @@ def rules__test_helper(test_case):
         )
         # If a `fixed` value is provided then check it matches
         if test_case.fix_str:
-            assert res == test_case.fix_str
+            assert res == test_case.fix_str, "result of fix doesn't match the fix_str"
+            assert_rule_pass_in_sql(
+                test_case.rule,
+                test_case.fix_str,
+                configs=test_case.configs,
+            )
```

Comment:
This assertion was the original idea of this PR, but is tricky, because there are some tests where the fix result also fails against the rule. Need to find find a way to explicitly record the exceptional expectation for those cases.

Reply:
Note to self: this assertion can be eventually enabled once all the existing cases that would fail on this have been migrated to assert the remaining violations explicitly as supported by https://github.com/sqlfluff/sqlfluff/pull/2718. 

The assertion to be added eventually is:

_if `violations_after_fix` is not given, `fix_str` must not have any remaining violations_